image_similarity_measures/evaluate.py

```diff
@@ -1,105 +1,63 @@
+def read_image(path):
+    logger.info("Reading image %s", os.path.basename(path))
+    if path.endswith(".tif") or path.endswith(".tiff"):
+        return np.rollaxis(rio.open(path).read(), 0, 3)
+    return cv2.imread(path)
 
-    if mode == "png":
-        logging.info("Reading image %s", Path(org_img_path).stem)
-        org_img = read_png(org_img_path)
-        logging.info("Reading image %s", Path(pred_img_path).stem)
-        pred_img = read_png(pred_img_path)
 
+def evaluation(org_img_path, pred_img_path, metrics):
+    output_dict = {}
+    org_img = read_image(org_img_path)
+    pred_img = read_image(pred_img_path)
 
-    out_value = eval(f"{metric}(org_img, pred_img)")
-    logger.info(f"{metric.upper()} value is: {out_value}")
-    if write_to_file:
-        metric_dict[metric] = {f"{metric.upper()}": out_value}
-        write_final_dict(metric, metric_dict)
+    for metric in metrics:
+        metric_func = metric_functions[metric]
+        out_value = float(metric_func(org_img, pred_img))
+        logger.info(f"{metric.upper()} value is: {out_value}")
+        output_dict[metric] = out_value
+    return output_dict
 
 
 def main():
+    logging.basicConfig(
+        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+        level=logging.INFO,
+    )
+    all_metrics = sorted(metric_functions.keys())
     parser = argparse.ArgumentParser(description="Evaluates an Image Super Resolution Model")
-    parser.add_argument("--org_img_path", type=str, help="Path to original input image")
-    parser.add_argument("--pred_img_path", help="Path to predicted images")
-    parser.add_argument("--metric", type=str, default="psnr", help=("use psnr, ssim, fsim, issm, uiq,"
-                                                                   " sam, sre or rmse as evaluation metric"))
-    parser.add_argument("--mode", type=str, default="tif", help="format of image, use either tif, or png, or jpg")
-    parser.add_argument("--write_to_file", action="store_true", help="final output will be written to a file.")
+    parser.add_argument("--org_img_path", help="Path to original input image", required=True, metavar="FILE")
+    parser.add_argument("--pred_img_path", help="Path to predicted image", required=True, metavar="FILE")
+    parser.add_argument("--metric", dest="metrics", action="append",
+                        choices=all_metrics + ['all'], metavar="METRIC",
+                        help="select an evaluation metric (%(choices)s) (can be repeated)")
     args = parser.parse_args()
+    if not args.metrics:
+        args.metrics = ["psnr"]
+    if "all" in args.metrics:
+        args.metrics = all_metrics
+
+    metric_values = evaluation(
+        org_img_path=args.org_img_path,
+        pred_img_path=args.pred_img_path,
+        metrics=args.metrics,
+    )
+    result_dict = {"image1": args.org_img_path, "image2": args.pred_img_path, "metrics": metric_values}
+    print(json.dumps(result_dict, sort_keys=True))
```

Comment:
For consistency better to use logger here as well.

Reply:
No, this is `print()` on purpose – `logger` writes to the stderr stream with formatting (times, log level etc.).

This will print machine-readable JSON to the stdout stream without any embellishments, so it's (as mentioned in the readme) easy to e.g. pipe into other apps.